lib/es6/toolset.js

```diff
@@ -60,8 +60,14 @@ Toolset.prototype.initializePosix = function (install) {
         this.cCompilerPath = "gcc";
     }
 
+    if( this.options.generator ){
+        if (install) {
+            this.log.info("TOOL", "Using " + this.options.generator + " generator, as specified from commandline.");
+        }
+        this.generator = this.options.generator;
+    }
     // 2: Generator
-    if (environment.isOSX) {
+    else if (environment.isOSX) {
```

Comment:
I figured it was pretty silly to do any sort of work to determine a default if it was already specified externally.